sw/device/tests/BUILD

```diff
@@ -358,6 +358,7 @@ opentitan_functest(
         "//sw/device/lib/base:macros",
         "//sw/device/lib/base:memory",
         "//sw/device/lib/base:mmio",
+        "//sw/device/lib/dif:flash_ctrl",
```

Comment:
@a-will : Just an FYI, looks like this test is using both the flash_ctrl.c and dif_flash_ctrl.c  (since I know you said you were in the process of removing `flash_ctrl.c`.